openfisca_france/model/prestations/agepi.py

```diff
@@ -1,54 +1,412 @@
+
+        #  Diminution de la précision car la comparaison : 14.77 <= 14.77 me renvoyait un False
+
+        epsilon = 0.0001
+
+        #  1
+        parents_isoles = individu.famille('nb_parents', period) == 1
+
+        #  2
+        condition_nb_enfants = individu.famille('agepi_nbenf', period) > 0
+
+        #  3
+        agepi_non_percues = not_(individu('agepi_percue_12_derniers_mois', period))
+
+        #  4
+        pe_categorie_demandeur_emploi = individu('pole_emploi_categorie_demandeur_emploi', period)
+
+        stagiaire_formation_professionnelle = individu('stagiaire', period)
+        contrat_aide = individu('en_contrat_aide', period)
+
+        categorie_4 = pe_categorie_demandeur_emploi == TypesCategoriesDemandeurEmploi.categorie_4
+        categorie_4_stagiaire_formation_professionnelle = categorie_4 * stagiaire_formation_professionnelle
 
-    def formula(famille, period, parameters):
-        isole = famille('nb_parents', period) == 1
-        nb_heures = famille.sum(famille.members('agepi_temps_travail_semaine', period), role = Famille.PARENT)
-        nbenf = famille('pe_nbenf', period)
+        categorie_5 = pe_categorie_demandeur_emploi == TypesCategoriesDemandeurEmploi.categorie_5
+        categorie_5_contrat_aide = categorie_5 * contrat_aide
+
+        categories_eligibles = ((pe_categorie_demandeur_emploi == TypesCategoriesDemandeurEmploi.categorie_1)
+                                + (pe_categorie_demandeur_emploi == TypesCategoriesDemandeurEmploi.categorie_2)
+                                + (pe_categorie_demandeur_emploi == TypesCategoriesDemandeurEmploi.categorie_3)
+                                + (categorie_4_stagiaire_formation_professionnelle + categorie_5_contrat_aide))
+
+        #  5
+        lieux_activite_eligibles = not_(individu('lieu_emploi_ou_formation', period) == TypesLieuEmploiFormation.non_renseigne)
+
+        #  6
+        contrat_de_travail_debut = individu('contrat_de_travail_debut', period)  # numpy.datetime64
+        contrat_de_travail_debut_en_mois = contrat_de_travail_debut.astype('M8[M]')
+
+        date_demande_limite = min_(
+            (contrat_de_travail_debut_en_mois + 1) + (contrat_de_travail_debut - contrat_de_travail_debut_en_mois),
+            (contrat_de_travail_debut_en_mois + 2) - np.timedelta64(1, 'D')
+            )
+
+        agepi_date_de_demande = individu("agepi_date_demande", period)
+        dates_demandes_agepi_eligibles = agepi_date_de_demande <= date_demande_limite
+
+        #  7
+        mayotte = individu.menage('residence_mayotte', period)
+        hors_mayotte = not_(mayotte)
+
+        allocation_individu = individu('allocation_retour_emploi', period)
+
+        allocation_minimale_hors_mayotte = parameters(period).chomage.allocation_retour_emploi.montant_minimum_hors_mayotte * hors_mayotte
+        allocation_minimale_mayotte = parameters(period).chomage.allocation_retour_emploi.montant_minimum_mayotte * mayotte
```

Comment:
Factoriser l'appel aux paramètres de la période ? 
```suggestion
        parametres_are = parameters(period).chomage.allocation_retour_emploi
        allocation_minimale_hors_mayotte = parametres_are.montant_minimum_hors_mayotte * hors_mayotte
        allocation_minimale_mayotte = parametres_are.montant_minimum_mayotte * mayotte
```

Reply:
De même pour les autres appels aux paramètres employant une portion commune de chemin. 🙂